### PR TITLE
feat(performance): prevent dashboards from being refreshed more often than every 5 minutes

### PR DESCRIPTION
**Pull request by yakkomajuri**
Getting started on the work for #14143.

We'll be moving to handling things at the API level but need to clarify things a bit beforehand, so just getting started with a simple change that should be non-controversial and at least prevent yesterday's event from happening again.

----

For the record, what I'm thinking going forward is:

Given the concept of a dashboard refresh doesn't exist in the backend (we just do `refresh=true` for the associated insights), I'm thinking we'll issue a refresh request to the backend for all insights, which will then decide if they should be refreshed based on dynamic settings (i.e. not 3min for all, but something "smarter"). We then need be clear on the UI which insights we did and which we didn't refresh e.g. "Did not refresh insight since it was already refreshed 4min ago".

### CODE REVIEW

**Comment by yakkomajuri** [comment]
How should I fix this?
```
frontend/src/scenes/dashboard/dashboardLogicType.ts(1093,39): error TS2503: Cannot find namespace 'dayjs'.
```

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 1]
```diff
@@ -672,6 +674,17 @@ export const dashboardLogic = kea<dashboardLogicType>([
                 return candidateShortest?.isValid() ? candidateShortest : null
             },
         ],
+        blockRefresh: [
+            (s) => [s.lastRefreshed],
```
Comment (pauldambra) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:678]: should we persist this value so a refresh of the page doesn't let you refresh (although the canny user could delete the value from storage, but if they're that determined 🤷 )

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 2]
```diff
@@ -672,6 +674,17 @@ export const dashboardLogic = kea<dashboardLogicType>([
...
+            (lastRefreshed) => {
+                return (
+                    !!lastRefreshed &&
+                    dayjs()
```
Comment (pauldambra) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:682]: we're importing `import { dayjs, now } from 'lib/dayjs'` we have a custom wrapper around dayjs (IIRC to control plugin init)

I think you have to use `now()` not `dayjs()` (but haven't checked)

**Review by pauldambra — APPROVED** [review]
Other than waiting 5 minutes to be sure it works 🚢

File: frontend/src/scenes/dashboard/dashboardLogic.tsx (hunk 1 above)
Comment (yakkomajuri) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:678]: this is based on `lastRefreshed` which is derived from the `last_refresh` value from insights that comes the backend so should hold across page refreshes, right?

Comment (pauldambra) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:678]: Yep, I read just the diff and made bad assumptions 🫠

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 3]
```diff
@@ -61,6 +61,8 @@ export const BREAKPOINT_COLUMN_COUNTS: Record<DashboardLayoutSize, number> = { s
 export const MIN_ITEM_WIDTH_UNITS = 3
 export const MIN_ITEM_HEIGHT_UNITS = 5
 
+export const DASHBOARD_REFRESH_INTERVAL_MINUTES = 5
```
Comment (macobo) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:64]: Should we call this `DASHBOARD_MIN_MANUAL_REFRESH_INTERVAL_MINUTES`?

Comment (yakkomajuri) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:64]: yup that sounds better - I just copied the name from the insights codebase and prepended with dashboard

**Comment by yakkomajuri** [comment]
Any thoughts on how I should fix the code quality error here @mariusandra ? I didn't even import anything new

**Comment by pauldambra** [comment]
@yakkomajuri kea typegen wasn't sure where to get dayjs and had picked a private dayjs for some reason.

I changed usage to be really specific about what was being returned/used and I think we'll pick the right one now

**Pull request merged**